scripts/assemble_changelog.py

```diff
@@ -100,21 +105,23 @@ def read_main_file(self, input_stream):
         of the class and may not act sensibly on an object that is already
         partially populated.
         """
+        # Parse the first level-2 section. Everything before the first
+        # level-3 section title ("###...") following the first level-2
+        # section title ("##...") is passed through as the header
+        # and everything after the second level-2 section title is passed
+        # through as the trailer. Inside the first level-2 section,
+        # split out the level-3 sections.
         level_2_seen = 0
         current_section = None
         for line in input_stream:
             level, content = self.title_level(line)
             if level == 2:
                 level_2_seen += 1
-                if level_2_seen <= 1:
-                    self.header.append(line)
-                else:
-                    self.trailer.append(line)
             elif level == 3 and level_2_seen == 1:
                 current_section = content
                 self.section_content.setdefault(content, [])
-            elif level_2_seen == 1 and current_section is not None:
-                if line.strip():
+            if level_2_seen == 1 and current_section is not None:
```

Comment:
Minor: perhaps a blank line before this `if` would be useful to visually separate the two `if`-`elif` structures.